src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -415,16 +422,50 @@ export class ObjectExplorerService implements IObjectExplorerService {
 						nodeProviders = nodeProviders.sort((a, b) => a.group!.toLowerCase().localeCompare(b.group!.toLowerCase()));
 						allProviders.push(...nodeProviders);
 					}
+					this.logService.trace(`${session.sessionId}: got providers for node expansion: ${allProviders.map(p => p.providerId).join(', ')}`);
+
+					const resolveExpansion = () => {
+						resolve(self.mergeResults(allProviders, resultMap, node.nodePath));
+						// Have to delete it after get all responses otherwise couldn't find session for not the first response
+						clearTimeout(expansionTimeout);
+						if (newRequest) {
+							delete self._sessions[session.sessionId!].nodes[node.nodePath];
+							this.logService.trace(`Deleted node ${node.nodePath} from session ${session.sessionId}`);
+						}
+					}
+
+					// Incase node status not found.
+					this._onNodeExpandedError.event(e => {
+						resultMap.set(e.providerId, e);
+						// When get all responses from all providers, merge results
+						if (resultMap.size === allProviders.length) {
+							//resolveExpansion();
```

Comment:
Still working on this?

Reply:
Sorry, accidentally pushed code for capturing timeout scenario